src/harvest_metrics.py

```diff
@@ -51,6 +56,13 @@ def failed_request(self, error_msg):
         )
 
     def submit(self):
+        if self.is_array:
+            return self.submit_array_metrics(self)
```

Comment:
Passing self into the function call is redundant here and will cause an error. Same goes with self.submit_single_metrics()

Reply:
yeah, I caught that... I didn't mean to push the code until I'd run it (sync instead of pull oops). Just was writing too quickly. It's fixed now.